### PR TITLE
Dynamic DynamoDB Table Fetching

### PR DESCRIPTION
**Pull request by amarjandu**
Add dynamic fetching of of dynamodb tables to health checks

Resolves: #2021

### CODE REVIEW

File: tests/test_health.py [hunk 1]
```diff
@@ -56,6 +56,16 @@ def test_full_health_check(self, mock_er, mock_ddb, mock_es):
...
 
+    @testmode.standalone
+    def test_resource_fetch(self):
+
+        service_tags = {"Key": "service", "Values": ["dss"]}
+        resource_list = health.get_resource_by_tag(resource_string='dynamodb:table', tag_filter=service_tags)
+        ddb_tables = [x['ResourceARN'].split('/')[1] for x in resource_list['ResourceTagMappingList'] if
+                      os.environ.get('DSS_DEPLOYMENT_STAGE') in x['ResourceARN']]
+        print(ddb_tables)
+        self.assertGreater(len(ddb_tables), 0)
```
Comment (xbrianh) [inline tests/test_health.py]: This test may be more future proof using `s3` instead of `dynamodb`.

File: dss/api/health.py [hunk 2]
```diff
@@ -33,10 +40,10 @@ def _get_dynamodb_status():
     """Checks dynamoDB table status, tables are explicitly specified within the function"""
     db_status = True
     stage = os.getenv("DSS_DEPLOYMENT_STAGE")
-    ddb_tables = ["dss-async-state-{}".format(stage),
-                  "dss-subscriptions-v2-aws-{}".format(stage),
-                  "dss-subscriptions-v2-gcp-{}".format(stage)
-                  ]
+    service_tags = {"Key": "service", "Values": ["dss"]}
+    resource_list = get_resource_by_tag(resource_string='dynamodb:table', tag_filter=service_tags)
```
Comment (xbrianh) [inline dss/api/health.py]: I think the same effect can be achieved replacing `dynamodb:table` with `dynamodb`.

File: tests/test_health.py (hunk 1 above)
Comment (amarjandu) [reply tests/test_health.py]: Updated test to work with s3

File: dss/api/health.py (hunk 2 above)
Comment (amarjandu) [reply dss/api/health.py]: I'm going to keep it as tables to ensure the resource filter is kept to tables in this function.

**Pull request merged**